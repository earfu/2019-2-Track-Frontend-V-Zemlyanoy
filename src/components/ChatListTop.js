import React from 'react';
import PropTypes from 'prop-types';

function ChatListTop({ name, onOpenMenu }) {
	return (
		<div className="chat-top-area">
			<button
				className="chat-bars-button"
				type="button"
				onClick={onOpenMenu}
				label="bars"
			>
				<i className="fa fa-bars" />
			</button>
			<p className="chat-top-name">{name}</p>
		</div>
	);
}

ChatListTop.propTypes = {
	name: PropTypes.string.isRequired,
	onOpenMenu: PropTypes.func.isRequired,
};

export default ChatListTop;
