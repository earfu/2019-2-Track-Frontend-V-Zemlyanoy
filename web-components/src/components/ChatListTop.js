import chatDefaults from '../chatDefaults';

const template = document.createElement('template');
template.innerHTML = `
    <style>
        .chat-top-area {
            font: 12pt bold;
            border: 1px solid black;
            border-radius: 20px;
            background-color: #0000ff;
            width: 100%;
            height: calc(100% - 6px);
            margin-top: 2px;
            text-align: center;
            display: flex;
            flex-direction: row;
        }
        #top-name {
            color: white;
            margin-left: 4px;
        }

        div button {
            background-color: #0000ff;
            border: none;
            border-radius: 20px;
            height: 100%;
            width: 60px;
            font-size: 16pt;
            margin-left: 4px;
        }

        button img {
            object-fit: cover;
            width: 100%;
            height: 100%;
        }
    </style>
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

    <div class="chat-top-area">
        <button class="bars-button" type="button" onclick=""><i class="fa fa-bars"></i></button>
        <p id="top-name"></p>
    </div>
`;

export default class ChatListTop extends HTMLElement {
  constructor() {
    super();
    /* this.shadowRoot = */this.attachShadow({ mode: 'open' });
    this.shadowRoot.appendChild(template.content.cloneNode(true));

    this.$name = this.shadowRoot.querySelector('#top-name');
    this.$name.textContent = chatDefaults.authorName; // default at creation

    this.$button = this.shadowRoot.querySelector('.bars-button');
    // this.$button.addEventListener('click', .bind(this));
  }

  setName(name) {
    this.$name.textContent = name;
  }
}

customElements.define('chat-list-top', ChatListTop);
