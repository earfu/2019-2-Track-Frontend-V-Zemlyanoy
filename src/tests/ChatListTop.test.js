import React from 'react';
import renderer from 'react-test-renderer';
import ChatListTop from '../components/ChatListTop';

it('renders correctly', () => {
  const component = renderer
    .create(<ChatListTop name="username" onOpenMenu={() => {}} />)
    .toJSON();
  // empty function used; a warning here is fine, function calls are tested in Storybook
  expect(component).toMatchSnapshot();
});
