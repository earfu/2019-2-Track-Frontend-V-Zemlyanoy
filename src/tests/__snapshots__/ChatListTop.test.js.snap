// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="chat-top-area"
>
  <button
    className="chat-bars-button"
    label="bars"
    onClick={null}
    type="button"
  >
    <i
      className="fa fa-bars"
    />
  </button>
  <p
    className="chat-top-name"
  >
    username
  </p>
</div>
`;
